src/legacy/core_plugins/kibana/public/dashboard/np_ready/lib/sync_app_filters.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import { filter, map } from 'rxjs/operators';
+import {
+  FilterManager,
+  esFilters,
+  compareFilters,
+  COMPARE_ALL_OPTIONS,
+} from '../../../../../../../plugins/data/public/';
+import { DashboardStateManager } from '../dashboard_state_manager';
+
+export function startSyncingAppFilters(
+  filterManager: FilterManager,
+  dashboardStateManager: DashboardStateManager
```

Comment:
could we make this generic so the same utility could be reused in visualize and discover ?

Reply:
Decoupled it from DashboardStateManager 👍